dsc_lib/src/parser/expressions.rs

```diff
@@ -32,64 +38,120 @@ impl Expression {
+                        };
+                        match index_value.kind() {
+                            "number" => {
+                                let value = index_value.utf8_text(statement_bytes)?;
+                                let value = serde_json::from_str(value)?;
+                                Accessor::Index(value)
+                            },
+                            "expression" => {
+                                return Err(DscError::Parser("Expression index not supported".to_string()));
+                            },
+                            _ => {
+                                return Err(DscError::Parser(format!("Invalid accessor kind: '{accessor_kind}'")));
+                            },
+                        }
+                    },
+                    _ => {
+                        return Err(DscError::Parser(format!("Invalid accessor kind: '{accessor_kind}'")));
+                    },
+                };
+                accessors.push(value);
             }
-            Some(result)
         }
-        else {
-            None
-        };
+
         Ok(Expression {
             function,
-            member_access,
+            accessors,
         })
     }
 
     /// Invoke the expression.
     ///
+    /// # Arguments
+    ///
+    /// * `function_dispatcher` - The function dispatcher to use.
+    /// * `context` - The context to use.
+    ///
+    /// # Returns
+    ///
+    /// The result of the expression.
+    ///
     /// # Errors
     ///
     /// This function will return an error if the expression fails to execute.
     pub fn invoke(&self, function_dispatcher: &FunctionDispatcher, context: &Context) -> Result<Value, DscError> {
         let result = self.function.invoke(function_dispatcher, context)?;
         trace!("Function result: '{:?}'", result);
-        if let Some(member_access) = &self.member_access {
-            debug!("Evaluating member access '{:?}'", member_access);
-            if !result.is_object() {
-                return Err(DscError::Parser("Member access on non-object value".to_string()));
-            }
-
+        if self.accessors.is_empty() {
+            Ok(result)
+        }
+        else {
+            debug!("Evaluating accessors");
             let mut value = result;
-            for member in member_access {
-                if !value.is_object() {
-                    return Err(DscError::Parser(format!("Member access '{member}' on non-object value")));
-                }
-
-                if let Some(object) = value.as_object() {
-                    if !object.contains_key(member) {
-                        return Err(DscError::Parser(format!("Member '{member}' not found")));
-                    }
-
-                    value = object[member].clone();
+            for accessor in &self.accessors {
+                match accessor {
+                    Accessor::Member(member) => {
+                        if !value.is_object() {
```

Comment:
Would it be less redundant to handle the non-object case as part the `as_object()` cast?

Reply:
yes, I think it can just be an `else` statement